### PR TITLE
feat: Add dumpgenesis subcommand

### LINKED ISSUE #5833: Add dumpgenesis subcommand
### Describe the feature

geth ref:

```
NAME:
   geth dumpgenesis - Dumps genesis block JSON configuration to stdout

USAGE:
   geth dumpgenesis [command options] [arguments...]

DESCRIPTION:
   
   The dumpgenesis command prints the genesis configuration of the network preset
   if one is set.  Otherwise it prints the genesis from the datadir.

OPTIONS:
   ETHEREUM

   
    --datadir value                (default: /Users/Matthias/Library/Ethereum)
          Data directory for the databases and keystore
   
    --goerli                       (default: false)
          Görli network: pre-configured proof-of-authority test network
   
    --mainnet                      (default: false)
          Ethereum mainnet
   
    --sepolia                      (default: false)
          Sepolia network: pre-configured proof-of-work test network

```

## TODO
add new subcommand that dumps the genesis file

https://github.com/paradigmxyz/reth/blob/33c8f405fcccc16e52916d96d404d64a2605e0a1/bin/reth/src/cli/mod.rs#L134-L134

see also:

https://github.com/paradigmxyz/reth/blob/33c8f405fcccc16e52916d96d404d64a2605e0a1/crates/primitives/src/chain/spec.rs#L567-L572

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AbnerZheng**
Close https://github.com/paradigmxyz/reth/issues/5833

If no Dumpgenesis argument(like `--mainnet`) is provided, we simply use the chain argument.

```
./reth dumpgenesis --help
Dumps genesis block JSON configuration to stdout

Usage: reth dumpgenesis [OPTIONS]

Options:
      --goerli
          Görli network: pre-configured proof-of-authority test network

      --mainnet
          Ethereum mainnet

      --holesky
          Holesky network: pre-configured proof-of-stake test network

      --sepolia
          Sepolia network: pre-configured proof-of-work test network

      --chain <CHAIN_OR_PATH>
          The chain this node is running.
          Possible values are either a built-in chain or the path to a chain specification file.

          Built-in chains:
              mainnet, sepolia, goerli, holesky, dev

          [default: mainnet]

      --instance <INSTANCE>
          Add a new instance of a node.

          Configures the ports of the node to avoid conflicts with the defaults. This is useful for running multiple nodes on the same machine.

          Max number of instances is 200. It is chosen in a way so that it's not possible to have port numbers that conflict with each other.

          Changes to the following port numbers: - DISCOVERY_PORT: default + `instance` - 1 - AUTH_PORT: default + `instance` * 100 - 100 - HTTP_RPC_PORT: default - `instance` + 1 - WS_RPC_PORT: default + `instance` * 2 - 2

          [default: 1]

  -h, --help
          Print help (see a summary with '-h')

Logging:
      --log.stdout.format <FORMAT>
          The format to use for logs written to stdout

          [default: terminal]

          Possible values:
          - json:     Represents JSON formatting for logs. This format outputs log records as JSON objects, making it suitable for structured logging
          - log-fmt:  Represents logfmt (key=value) formatting for logs. This format is concise and human-readable, typically used in command-line applications
          - terminal: Represents terminal-friendly formatting for logs

      --log.stdout.filter <FILTER>
          The filter to use for logs written to stdout

          [default: ]

      --log.file.format <FORMAT>
          The format to use for logs written to the log file

          [default: terminal]

          Possible values:
          - json:     Represents JSON formatting for logs. This format outputs log records as JSON objects, making it suitable for structured logging
          - log-fmt:  Represents logfmt (key=value) formatting for logs. This format is concise and human-readable, typically used in command-line applications
          - terminal: Represents terminal-friendly formatting for logs

      --log.file.filter <FILTER>
          The filter to use for logs written to the log file

          [default: debug]

      --log.file.directory <PATH>
          The path to put log files in

          [default: /Users/abnerzheng/Library/Caches/reth/logs]

      --log.file.max-size <SIZE>
          The maximum size (in MB) of one log file

          [default: 200]

      --log.file.max-files <COUNT>
          The maximum amount of log files that will be stored. If set to 0, background file logging is disabled

          [default: 5]

      --log.journald
          Write logs to journald

      --log.journald.filter <FILTER>
          The filter to use for logs written to journald

          [default: error]

      --color <COLOR>
          Sets whether or not the formatter emits ANSI terminal escape codes for colors and other text formatting

          [default: always]

          Possible values:
          - always: Colors on
          - auto:   Colors on
          - never:  Colors off

Display:
  -v, --verbosity...
          Set the minimum log level.

          -v      Errors
          -vv     Warnings
          -vvv    Info
          -vvvv   Debug
          -vvvvv  Traces (warning: very verbose!)

  -q, --quiet
          Silence all log output
```

### CODE REVIEW

File: bin/reth/src/commands/dump_genesis.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    #[arg(long, verbatim_doc_comment)]
+    mainnet: bool,
+
+    /// Holesky network: pre-configured proof-of-stake test network
+    #[arg(long, verbatim_doc_comment)]
+    holesky: bool,
+
+    /// Sepolia network: pre-configured proof-of-work test network
+    #[arg(long, verbatim_doc_comment)]
+    sepolia: bool,
```
Comment (onbjerg) [inline bin/reth/src/commands/dump_genesis.rs:25]: please make `chain` compatible with this, i.e. remove these flags so you can just do `--chain mainnet`

**Review by onbjerg — CHANGES_REQUESTED** [review]
i would like this to be named dump-genesis or similar, and i would also like for the book to be updated: https://paradigmxyz.github.io/reth/cli/reth.html

i would also like a test

File: bin/reth/src/commands/dump_genesis.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    #[arg(
+    long,
+    value_name = "CHAIN_OR_PATH",
+    long_help = chain_help(),
+    default_value = SUPPORTED_CHAINS[0],
+    value_parser = genesis_value_parser
```
[the remark this answers is not in the document]
Comment (AbnerZheng) [reply bin/reth/src/commands/dump_genesis.rs]: done.

File: bin/reth/src/commands/dump_genesis.rs (hunk 1 above)
Comment (AbnerZheng) [reply bin/reth/src/commands/dump_genesis.rs:24]: I'm happy to make these changes. @mattsse WDYT

Comment (mattsse) [reply bin/reth/src/commands/dump_genesis.rs:24]: scratch that, this just complicates things

**Review by mattsse — CHANGES_REQUESTED** [review]
@AbnerZheng let's just keep the chain Arc Chainspec field and remove the others.

I think this is fine, and mirrors the behaviour of other subcommand. having geth compatibility here isn't that important.

so please remove all bool flags and just keep the chain field

and please add a parser test, see other subcommand for examples

**Comment by AbnerZheng** [comment]
Updated help message:
```
❯ ./reth dump-genesis --help
Dumps genesis block JSON configuration to stdout

Usage: reth dump-genesis [OPTIONS]

Options:
      --chain <CHAIN_OR_PATH>
          The chain this node is running.
          Possible values are either a built-in chain or the path to a chain specification file.

          Built-in chains:
              mainnet, sepolia, goerli, holesky, dev

          [default: mainnet]

      --instance <INSTANCE>
          Add a new instance of a node.

          Configures the ports of the node to avoid conflicts with the defaults. This is useful for running multiple nodes on the same machine.

          Max number of instances is 200. It is chosen in a way so that it's not possible to have port numbers that conflict with each other.

          Changes to the following port numbers: - DISCOVERY_PORT: default + `instance` - 1 - AUTH_PORT: default + `instance` * 100 - 100 - HTTP_RPC_PORT: default - `instance` + 1 - WS_RPC_PORT: default + `instance` * 2 - 2

          [default: 1]

  -h, --help
          Print help (see a summary with '-h')

Logging:
      --log.stdout.format <FORMAT>
          The format to use for logs written to stdout

          [default: terminal]

          Possible values:
          - json:     Represents JSON formatting for logs. This format outputs log records as JSON objects, making it suitable for structured logging
          - log-fmt:  Represents logfmt (key=value) formatting for logs. This format is concise and human-readable, typically used in command-line applications
          - terminal: Represents terminal-friendly formatting for logs

      --log.stdout.filter <FILTER>
          The filter to use for logs written to stdout

          [default: ]

      --log.file.format <FORMAT>
          The format to use for logs written to the log file

          [default: terminal]

          Possible values:
          - json:     Represents JSON formatting for logs. This format outputs log records as JSON objects, making it suitable for structured logging
          - log-fmt:  Represents logfmt (key=value) formatting for logs. This format is concise and human-readable, typically used in command-line applications
          - terminal: Represents terminal-friendly formatting for logs

      --log.file.filter <FILTER>
          The filter to use for logs written to the log file

          [default: debug]

      --log.file.directory <PATH>
          The path to put log files in

          [default: /Users/abnerzheng/Library/Caches/reth/logs]

      --log.file.max-size <SIZE>
          The maximum size (in MB) of one log file

          [default: 200]

      --log.file.max-files <COUNT>
          The maximum amount of log files that will be stored. If set to 0, background file logging is disabled

          [default: 5]

      --log.journald
          Write logs to journald

      --log.journald.filter <FILTER>
          The filter to use for logs written to journald

          [default: error]

      --color <COLOR>
          Sets whether or not the formatter emits ANSI terminal escape codes for colors and other text formatting

          [default: always]

          Possible values:
          - always: Colors on
          - auto:   Colors on
          - never:  Colors off

Display:
  -v, --verbosity...
          Set the minimum log level.

          -v      Errors
          -vv     Warnings
          -vvv    Info
          -vvvv   Debug
          -vvvvv  Traces (warning: very verbose!)

  -q, --quiet
          Silence all log output
```

**Review by mattsse — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]
thanks!

**Pull request merged**